### ISSUE TITLE
`debug_model_outputs()` what is being checked?

### ISSUE THREAD

**Issue by petteriTeikari** [moment-timeseries-foundation-model/moment-research#13]
Hi,

i was trying to reproduce your finetuned anomaly detection for my own data and was wondering what was this debug function doing here?
https://github.com/moment-timeseries-foundation-model/moment-research/blob/3ab637e413f35f2c317573c0ace280d825c558de/moment/tasks/base.py#L341

I would think that you would like to check if `any()` of the gradients is not `isfinite()`? e.g. with `~` or did i misunderstood something here as got confused as I got flagged for illegal gradients by that function?

```
illegal_encoder_grads = (
        torch.stack(
            [~torch.isfinite(p).any() for p in model.encoder.parameters()]
        )
        .any()
        .item()
    )
```

**Comment by mononitogoswami** [comment]
Hi Petteri, 
Thanks for your interest in MOMENT!
I used this function to check for offending (NaN, infinite) gradients. This should generally not happen, if the time series has sufficient observed values (unmasked time steps) and optimization is done in full precision or brain float 16. 

Best,
Mononito

**Comment by petteriTeikari** [comment]
Yes I was wondering if it just had to be inverted the check?

I mean that threw a warning for me, but I got the model to train with loss coming down (while the actual adjusted f1 did not came out so great)
